buildchain/buildchain/vagrant.py

```diff
@@ -40,16 +42,53 @@ def mkdir_dot_vagrant() -> None:
         'uptodate': [True],
     }
 
+def task_vagrantupnoprov() -> types.TaskDict:
+    """Run `vagrant up` without provisioning a development environment."""
+    vagrantupnoprov = [config.VAGRANT, 'up']
+    vagrantupnoprov.extend(map(shlex.quote,
+        config.VAGRANT_UP_ARGS + tuple(('--no-provision',))))
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantupnoprov))],
+        'file_dep': [constants.VAGRANT_SSH_KEY_PAIR],
+        'task_dep': [],
+        'uptodate': [False],
+    }
+
+def task_vagrantsnapshot() -> types.TaskDict:
+    """Snapshot development environment."""
+    vagrantsnap = [config.VAGRANT, 'snapshot', 'save', '--force', 'bootstrap', config.VAGRANT_SNAPSHOT_NAME]
+
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantsnap))],
+        'file_dep': [],
+        'task_dep': ['vagrantupnoprov'],
+        'uptodate': [run_once]
```

Comment:
Maybe good for now, but could be better to check if the snapshot exist in vagrant (by checking the output of `vagrant snapshot list`).

You can create a debt ticket for this one, no need to tackle it here.

Reply:
#971